src/pkg/audit/dao/dao.go

```diff
@@ -34,15 +39,85 @@ type DAO interface {
 	Get(ctx context.Context, id int64) (access *model.AuditLog, err error)
 	// Delete the audit log specified by ID
 	Delete(ctx context.Context, id int64) (err error)
+	// Purge the audit log
+	Purge(ctx context.Context, retentionHour int, includeOperations []string, dryRun bool) (int64, error)
 }
 
 // New returns an instance of the default DAO
 func New() DAO {
 	return &dao{}
 }
 
+var allowedMaps = map[string]string{
+	strings.ToLower(rbac.ActionPull.String()):   "",
+	strings.ToLower(rbac.ActionCreate.String()): "",
+	strings.ToLower(rbac.ActionDelete.String()): "",
+}
+
 type dao struct{}
 
+// Purge delete expired audit log
+func (*dao) Purge(ctx context.Context, retentionHour int, includeOperations []string, dryRun bool) (int64, error) {
```

Comment:
is it possible to have two methods here? Then, let the job to select which one is for dry-run.